modules/appeal/src/main/AppealApi.scala

```diff
@@ -83,10 +91,36 @@ final class AppealApi(
     coll.update.one($id(appeal.id), appeal.toggleMute).void
 
   def setReadById(userId: User.ID) =
-    coll.byId[Appeal](userId) flatMap { _ ?? setRead }
+    byId(userId) flatMap { _ ?? setRead }
 
   def setUnreadById(userId: User.ID) =
-    coll.byId[Appeal](userId) flatMap { _ ?? setUnread }
+    byId(userId) flatMap { _ ?? setUnread }
 
   def onAccountClose(user: User) = setReadById(user.id)
+
+  def snooze(mod: User, appealId: User.ID, duration: String): Funit =
+    byId(appealId) flatMap { (appeal: Option[Appeal]) =>
+      for {
+        a <- appeal
+        d <- Snooze.Duration(duration)
+      } yield snoozer.set(mod.id, a.id, d)
+      funit
+    }
```

Comment:
There may be a better way to do that

Reply:
use `foreach` for side effecting (instead of `flatMap` which is about composing futures)